File: 2P/POO/Lista_Basica/Exercicio01/src/Ponto.java
/* Objetivo:
 * Autor   : William Silva (https://gitlab.com/unclWill)
 * Data    : 26/02/2024
 * Exercício 1) Crie um programa que calcule a distância entre dois pontos no plano cartesiano.
 */

public class Ponto {
    // Um ponto tem as coordenadas A e B.
    private double x;
    private double y;
    public Ponto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        if (!Double.isNaN(x)) {
            this.x = x;
        }
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        if (!Double.isNaN(y)) {
            this.y = y;
        }
    }
}

File: 1P/FP/Exercicios_Algoritmos/FP09_S43_E06/CalcDistPts/Ponto.cs
/* Objetivo : Calcular distância entre dois pontos.
 * Autor    : William Silva (https://github.com/unclWill)
 * Data     : 07/11/2023
 * Material : FP09 (Classes)
 * Slide    : 43
 * Exercício: 06
 */

/* 6) Crie uma classe chamada ponto contendo apenas as coordenadas x e y (inteiros) do ponto. 
 * Declare 2 pontos, leia as coordenadas x e y de cada um e calcule a distância entre eles. 
 * Apresente no final a distância entre os dois pontos.
 */

using System;

namespace CalcDistPts
{
    public class Ponto
    {
        private int x;
        private int y;

        public int X
        {
            get { return x; }
            set { x = value; }
        }

        public int Y
        {
            get { return y; }
            set { y = value; }
        }
    }
}